### PR TITLE
[MOB-532] Reduce time of payment dialog

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

   Reduces time of the payment dialog's loading 

**Database changed?**

No

**Where should the reviewer start?**

PaymentMethodsPresenter.kt

**How should this be manually tested?**

Check if nothing breaks on the different payment flows (inapp and onestep)

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/MOB-532

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: bdsbilling/src/main/java/com/appcoins/wallet/bdsbilling/WalletService.kt [hunk 1]
```diff
@@ -5,7 +5,10 @@ import io.reactivex.Single
...
 interface WalletService {
 
-    fun getWalletAddress(): Single<String>
+  fun getWalletAddress(): Single<String>
 
-    fun signContent(content: String): Single<String>
+  @Deprecated("Use signContent() instead to avoid duplicated code")
+  fun signContent(content: String): Single<String>
+
+  fun signContent(): Single<WalletAddressModel>
```
Comment (FredericoSantinho) [inline bdsbilling/src/main/java/com/appcoins/wallet/bdsbilling/WalletService.kt]: shouldn't this have a different name? It doesn't sign a content, it signs the wallet address, right?

File: app/src/main/java/com/asfoundation/wallet/ui/Erc681ReceiverPresenter.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ public void present(Bundle savedInstanceState) {
     return throwable instanceof WalletNotFoundException ? createWallet() : Single.error(throwable);
   }
 
-  public void pause() {
+  void pause() {
```
Comment (FredericoSantinho) [inline app/src/main/java/com/asfoundation/wallet/ui/Erc681ReceiverPresenter.java:62]: Shouldn't the method present() be as well package protected for the sake of consistency?

Comment (JDSM01) [reply app/src/main/java/com/asfoundation/wallet/ui/Erc681ReceiverPresenter.java:62]: Agree, but the Ide didn't warn me about that XD

File: app/src/main/java/com/asfoundation/wallet/ui/iab/IabActivity.kt [hunk 3]
```diff
@@ -258,9 +234,7 @@ class IabActivity : BaseActivity(), IabView, UriNavigator {
 
   override fun onSaveInstanceState(outState: Bundle) {
     super.onSaveInstanceState(outState)
-
-    outState.putBundle(SKU_DETAILS, skuDetails)
-    outState.putBoolean(FIRST_IMPRESSION, firstImpression)
+    presenter.onSaveInstance(outState)
```
Comment (FredericoSantinho) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/IabActivity.kt:237]: Does it make sense that presenter knows about the Bundle? Doesn't it violate Clean Architecture's Dependency Rule?

File: app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsView.kt [hunk 4]
```diff
@@ -19,12 +19,12 @@ interface PaymentMethodsView {
   fun finish(bundle: Bundle)
   fun showLoading()
   fun hideLoading()
-  fun getCancelClick(): Observable<PaymentMethod>
+  fun getCancelClick(): Observable<PaymentMethod?>
```
Comment (FredericoSantinho) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsView.kt]: How can the PaymentMethod be nullable if Observable cannot emit null values? Same for the next two changes.

**Pull request merged**